src/interface.js

```diff
@@ -0,0 +1,160 @@
+'use strict';
+
+$(document).ready(function(){
+    var game;
+    var roll = 1;
+
+    createSelector(roll, 10);
+
+    $("#submit_score").on("click", function() {
+        event.preventDefault();
+        game = new Game();
+        game.compute(getScoresArray());
+        showOutput();
+        $(this).prop("disabled",true);
+    });
+
+    $("#reset").on("click", function() {
+        location.reload();
+    });
+
+    $(document.body).on('change', 'select', function() {
+        if (roll > 19) {
+            updateLastSelectors();
+        } else {
+            updateRollSelectors(rollScore(roll));
+        }
+    });
+
+// Helpers -------------------------------------------- Helpers //
+
+    function createSelector(id_number, limit) {
```

Comment:
jQuery is generally going to end up like this – even good jQuery. That is: a nice-ish interface and a whole bunch of helpers.

It can be refactored by using one of the framework architectures you know (i.e. MVC) and building a controller that uses jQuery methods, but it'll probably end up much the same.

An alternative thing to do is create a 'view model' – that is, Objects that use jQuery in their methods. You identify objects in the view just like you do in the rest of your program: what nouns are you using? I see: `SpiritBowler`, `Total`, `FrameScore`. If you move the logic for rendering and updating each of these 'view objects' into their own object, you start moving towards a _component-based_ architecture.